src/tasks/python/PythonExtensionHelper.ts

```diff
@@ -0,0 +1,116 @@
+                    }
+
+                    const maxRetriesCount = 20;
+                    let retries = 0;
+                    let created = false;
+
+                    // Look for the magic string below in the log file with a retry every 0.5 second for a maximum of 10 seconds.
+                    // TODO: Should be gone as soon as the retry logic is part of the Python debugger/extension.
+                    while (++retries < maxRetriesCount && !created) {
+                        if (fse.existsSync(debuggerSemaphorePath)) {
+                            const contents = fse.readFileSync(debuggerSemaphorePath);
+
+                            created = contents.toString().indexOf("Starting server daemon on") >= 0;
+                            if (created) {
+                                break;
+                            }
+                        }
+
+                        await delay(500);
+                    }
+
+                    if (created) {
+                        resolve();
+                    } else {
+                        reject("Failed to attach the debugger within the alotted timeout.");
+                    }
+                }});
+        })
+    }
+
+    export function getRemotePtvsdCommand(target: PythonFileTarget | PythonModuleTarget, args?: string[], options?: DebugLaunchOptions): string {
+        let fullTarget: string;
+
+        if ((target as PythonFileTarget).file) {
+            fullTarget = (target as PythonFileTarget).file;
+        } else if ((target as PythonModuleTarget).module) {
+            fullTarget = `-m ${(target as PythonModuleTarget).module}`;
+        } else {
+            throw new Error("One of either module or file must be provided.");
+        }
+
+        options = options || {};
+        options.host = options.host || "0.0.0.0";
+        options.port = options.port || PythonDefaultDebugPort;
+        options.wait = !!options.wait;
+        args = args || [];
+
+        return `/pydbg/ptvsd --host ${options.host} --port ${options.port} ${options.wait ? "--wait" : ""} ${fullTarget} ${args.join(" ")}`;
+    }
+
+    export function getLauncherFolderPath(): string {
+        const pyExt = vscode.extensions.getExtension("ms-python.python");
+
+        if (!pyExt) {
+            throw new Error("For debugging Python apps in a container to work, the Python extension must be installed.");
+        }
+
+        const debuggerPath = path.join(pyExt.extensionPath, "pythonFiles", "lib", "python");
+        const oldDebugger = path.join(debuggerPath, "old_ptvsd");
+        const newDebugger = path.join(debuggerPath, "new_ptvsd");
+
+        // Always favor the old_ptvsd debugger since it will work in all cases.
+        // If it is not found, then look for the new instead.
+        // TODO: This should be revisited when the Python extension releases the new debugger since it might have a different name.
+
+        if (fse.existsSync(oldDebugger)) {
```

Comment:
Can we use the async forms?

Reply:
Same comment about it accepting a callback and returns void instead of a Promise, plus for some reason "fse.exists()" is deprecated.

Reply:
Same as above, I think it's `fse.pathExists()`